src/cryojax/simulator/_optics.py

```diff
@@ -17,214 +16,176 @@
-    Compute a Contrast Transfer Function (CTF).
-
-    See ``cryojax.simulator.compute_ctf`` for more
-    information.
-
-    Attributes
-    ----------
-    degrees :
-    defocus_u :
-    defocus_v :
-    defocus_angle :
-    voltage :
-    spherical_aberration :
-    amplitude_contrast_ratio :
-    phase_shift :
+    """Compute the Contrast Transfer Function (CTF) in for a weakly
+    scattering specimen.
+
+    **Attributes:**
+
+    `defocus_u`: The major axis defocus in Angstroms.
+
+    `defocus_v`: The minor axis defocus in Angstroms.
+
+    `astigmatism_angle`: The defocus angle.
+
+    `voltage_in_kv`: The accelerating voltage in kV.
+
+    `spherical_aberration_in_mm`: The spherical aberration coefficient in mm.
+
+    `amplitude_contrast_ratio`: The amplitude contrast ratio.
+
+    `phase_shift`: The additional phase shift.
+
+    `degrees`: Whether or not the `astigmatism_angle` and `phase_shift` are given
+              in degrees or radians.
     """
 
     defocus_u: Real_ = field(default=10000.0, converter=jnp.asarray)
     defocus_v: Real_ = field(default=10000.0, converter=jnp.asarray)
-    defocus_angle: Real_ = field(default=0.0, converter=jnp.asarray)
-    voltage: Real_ = field(default=300.0, converter=jnp.asarray)
-    spherical_aberration: Real_ = field(default=2.7, converter=jnp.asarray)
-    amplitude_contrast: Real_ = field(default=0.1, converter=jnp.asarray)
+    astigmatism_angle: Real_ = field(default=0.0, converter=jnp.asarray)
+    voltage_in_kilovolts: Real_ = field(default=300.0, converter=jnp.asarray)
+    spherical_aberration_in_mm: Real_ = field(default=2.7, converter=jnp.asarray)
+    amplitude_contrast_ratio: Real_ = field(default=0.1, converter=jnp.asarray)
     phase_shift: Real_ = field(default=0.0, converter=jnp.asarray)
 
     degrees: bool = field(static=True, default=True)
 
+    @property
+    def wavelength_in_angstroms(self):
+        voltage_in_volts = 1000.0 * self.voltage_in_kilovolts  # kV to V
+        return 12.2643 / (voltage_in_volts + 0.97845e-6 * voltage_in_volts**2) ** 0.5
+
     def __call__(
         self, freqs: ImageCoords, defocus_offset: Real_ | float = 0.0
     ) -> RealImage:
-        return compute_ctf(
+        if self.degrees:  # degrees to radians
+            phase_shift = jnp.deg2rad(self.phase_shift)
+            astigmatism_angle = jnp.deg2rad(self.astigmatism_angle)
+        return _compute_ctf(
             freqs,
-            self.defocus_u + defocus_offset,
-            self.defocus_v + defocus_offset,
-            self.defocus_angle,
-            self.voltage,
-            self.spherical_aberration,
-            self.amplitude_contrast,
-            self.phase_shift,
-            degrees=self.degrees,
+            self.defocus_u + jnp.asarray(defocus_offset),
+            self.defocus_v + jnp.asarray(defocus_offset),
+            astigmatism_angle,
+            self.wavelength_in_angstroms,
+            self.spherical_aberration_in_mm * 1e7,  # mm to Angstroms
```

Comment:
it's a small thing, but I would document the comment as code:

spherical_aberration_in_angstroms = self.spherical_aberration_in_mm*1e7

and then use spherical_aberration_in_angstroms in the call

Reply:
comments are easier to loose track of

Reply:
Good point yes